src/features/transactionTable/components/columns/DeleteButton.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  AlertDialog,
+  AlertDialogAction,
+  AlertDialogCancel,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogTitle,
+  AlertDialogTrigger,
+} from '@/components/ui/alert-dialog';
+import { Button } from '@/components/ui/button';
+import { Trash as TrashIcon } from 'lucide-react';
+import { TransactionTableData } from '@/components/AddTransactionForm/AddTransactionForm';
+import { toast } from '@/hooks/useToast';
+import { useTransactionStore } from '@/stores/TransactionStore';
+
+type DeleteButtonProps = {
+  transactionToEdit: TransactionTableData;
+};
+
+export const DeleteButton = ({ transactionToEdit }: DeleteButtonProps) => {
+  const deleteTransaction = useTransactionStore(
+    (state) => state.deleteTransaction,
+  );
+
+  const handleDeleteTransaction = () => {
+    try {
+      deleteTransaction(transactionToEdit.id);
+
+      toast({
+        title: 'Transakce byla úspěšně smazána',
+        duration: 5000,
+        className: 'bg-green-100 border-green-500 text-green-900',
+      });
+    } catch (error) {
+      console.error('Error deleting transaction:', error);
+
+      toast({
+        variant: 'destructive',
+        title: 'Nastala chyba při mazání transakce',
+      });
+    }
+  };
```

Comment:
Toasts and the try catch block should be handled within the `deleteTransaction` function. The only thing that can throw an error is `localStorage.setItem()` so that should be inside the try catch block.